dev/Detours/Detours.vcxproj

```diff
@@ -250,7 +250,7 @@
       <FunctionLevelLinking>true</FunctionLevelLinking>
       <RemoveUnreferencedCodeData>false</RemoveUnreferencedCodeData>
       <OmitDefaultLibName>true</OmitDefaultLibName>
-      <BufferSecurityCheck>false</BufferSecurityCheck>
+      <AdditionalOptions>/ZH:SHA_256 %(AdditionalOptions)</AdditionalOptions>
```

Comment:
Can these changes follow the same advice as Jon gave [here](https://github.com/microsoft/ProjectReunion/pull/553/files#r593628407)?

>Please change this to use a model like:
>
>```
><AdditionalOptions>$(AdditionalPipelineLinkerOptions) %(AdditionalOptions)</AdditionalOptions>
>```
>
>Where that "additional pipeline" value is set by the ADO build pipeline as necessary. This lets us configure the linker commands during official build runs but not impact the individual developer's output.

Expect developers to add more *.vcxproj. Any common properties we want set should be handled in a similar way to better scale as the work (and #devs) grows.

Reply:
binskim is supposed to validate what we ship, which should be the default configuration for the build. This is different from some of the needs we have for one-off customized scanning builds.

Reply:
@BenJKuhn I'm confused. Or maybe I need more coffee

Are you agreeing this should be handled via `$(blah)` like @jonwis advocated?
Or no `/ZH:blah` etc here is the better answer?

Reply:
I think in this case, you want to push the better hashing practices here onto all devs (instead of injecting during CI time). This also makes it one baby step easier to check off some hashing algorithm use compliance checkboxes. (We deal with this stuff in government sector.)

Reply:
Agreed. We should strive to keep build configs identical between inner loop and pipeline builds as much as possible. Provided we don't impose an undue burden on inner loop developers. I don't think any of the options here impose such a burden.

Your comment about adding more projects and developers, though, makes me think that we should consider collecting some of these build configurations into a props/targets file. Of course, then we're paying a concept tax by moving some of these build definitions out of the local vcxproj files, so that devs have to go hunting to figure out why/where certain build options have been set. (Also, I'm not sure how we set debug vs release options in a props/targets file.)